src/Persistence/MassTransit.DynamoDbIntegration/DynamoDbIntegration/Saga/DynamoDbDatabaseContext.cs

```diff
@@ -46,37 +45,41 @@ public async Task<TSaga> Load(Guid correlationId)
         public async Task Update(SagaConsumeContext<TSaga> context)
         {
             var instance = context.Saga;
-            var updateLock = await Lock(instance, context.CancellationToken).ConfigureAwait(false);
             try
             {
-                instance.Version++;
-
-                var existingInstance = await Load(instance.CorrelationId).ConfigureAwait(false);
-                if (existingInstance.Version >= instance.Version)
-                    throw new DynamoDbSagaConcurrencyException("Saga version conflict", typeof(TSaga), instance.CorrelationId);
-
-                await Save(instance).ConfigureAwait(false);
+                var toUpdateSaga = new DynamoDbSaga
```

Comment:
Extract method to a common builder for this and the other method (Save). Just increment the version prior to building the dynamo db saga.